`csstype` bump to 3.1.3 causing BREAKING CHANGEs on type-level

While bumping griffel build dependencies within fluent core, I run into unexpected type errors.

I was able to troubleshoot it deeper and I can confirm that the culprit of this issue is `csstype` package bump from `3.1.2` to `3.1.3` which happened [in `@griffel/core@1.15.2` ( and related packages to that release )](https://github.com/microsoft/griffel/blob/main/packages/core/CHANGELOG.md#1152).

Also other folks run into this issue: https://github.com/frenic/csstype/issues/189

I dont have context if that package minor bump is required, but if not I'd recommend to revert to `3.1.2` until the package authors will release a fixed version.

**Reproduction:**


**Repro:**

checkout: https://github.com/microsoft/fluentui/pull/30694

`yarn workspace @fluentui/react-positioning type-check`

```
src/createArrowStyles.ts:93:34 - error TS2769: No overload matches this call.
  Overload 1 of 6, '(width: BorderWidthInput, style: BorderStyleInput, color: BorderColorInput): BorderBottomStyle', gave the following error.
    Argument of type 'BorderBottomWidth<GriffelStylesCSSValue> | readonly BorderBottomWidth<GriffelStylesCSSValue>[]' is not assignable to parameter of type 'BorderWidthInput'.
      Type 'readonly BorderBottomWidth<GriffelStylesCSSValue>[]' is not assignable to type 'BorderWidthInput'.
        The type 'readonly BorderBottomWidth<GriffelStylesCSSValue>[]' is 'readonly' and cannot be assigned to the mutable type 'BorderWidth<GriffelStylesCSSValue>[]'.
  Overload 2 of 6, '(style: BorderStyleInput, width: BorderWidthInput, color: BorderColorInput): BorderBottomStyle', gave the following error.
    Argument of type 'BorderBottomWidth<GriffelStylesCSSValue> | readonly BorderBottomWidth<GriffelStylesCSSValue>[]' is not assignable to parameter of type 'BorderStyleInput'.
      Type '0' is not assignable to type 'BorderStyleInput'.

93       ...shorthands.borderBottom(borderWidth, borderStyle, borderColor),
                                    ~~~~~~~~~~~
```

--- a/change/@griffel-style-types-f154bb98-0961-405f-ae62-cc5beb9eff4f.json
+++ b/change/@griffel-style-types-f154bb98-0961-405f-ae62-cc5beb9eff4f.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix: use custom fallback type for properties",
+  "packageName": "@griffel/style-types",
+  "email": "olfedias@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/core/src/shorthands/types.test.ts
+++ b/packages/core/src/shorthands/types.test.ts
@@ -1,0 +1,25 @@
+import { type GriffelStyle } from '@griffel/style-types';
+import { borderBottom } from './borderBottom';
+
+describe('types tests', () => {
+  it('borderBottom', () => {
+    type Options = {
+      width?: GriffelStyle['borderBottomWidth'];
+      style?: GriffelStyle['borderBottomStyle'];
+      color?: GriffelStyle['borderBottomColor'];
+    };
+    const options: Options = {
+      width: '2px',
+      style: 'solid',
+      color: 'red',
+    };
+
+    const { width = '1px', style = 'solid', color = 'red' } = options;
+
+    expect(borderBottom(width, style, color)).toEqual({
+      borderBottomWidth: '2px',
+      borderBottomStyle: 'solid',
+      borderBottomColor: 'red',
+    });
+  });
+});

--- a/packages/style-types/src/makeStyles.ts
+++ b/packages/style-types/src/makeStyles.ts
@@ -1,6 +1,6 @@
 import type * as CSS from 'csstype';
 
-import type { GriffelStylesCSSValue } from './shared';
+import type { Fallback, GriffelStylesCSSValue } from './shared';
 import type { GriffelStylesUnsupportedCSSProperties } from './unsupported-properties';
 
 //
@@ -9,7 +9,7 @@ import type { GriffelStylesUnsupportedCSSProperties } from './unsupported-proper
 //
 
 type GriffelStylesCSSProperties = Omit<
-  CSS.PropertiesFallback<GriffelStylesCSSValue>,
+  Fallback<CSS.Properties<GriffelStylesCSSValue>>,
   // We have custom definition for "animationName"
   'animationName'
 > &

--- a/packages/style-types/src/shared.ts
+++ b/packages/style-types/src/shared.ts
@@ -1,3 +1,4 @@
 export type GriffelStylesCSSValue = string | 0;
 
 export type ValueOrArray<T> = T | Array<T>;
+export type Fallback<T> = { [P in keyof T]: T[P] | NonNullable<T[P]>[] };

--- a/packages/style-types/src/shared.ts
+++ b/packages/style-types/src/shared.ts
@@ -1,8 +1,9 @@
 export type GriffelStylesCSSValue = string | 0;
 
-export type ValueOrArray<T> = T | Array<T>;
 /**
  * instead of using `CSS.PropertiesFallback` from csstype that remaps properties to `readonly` union we need to use our cutom Fallback mapped type in order to avoid type issues
  * @see https://github.com/frenic/csstype/blob/master/src/typescript.ts#L73
  */
 export type Fallback<T> = { [P in keyof T]: T[P] | NonNullable<T[P]>[] };
+
+export type ValueOrArray<T> = T | Array<T>;

--- a/packages/style-types/src/shared.ts
+++ b/packages/style-types/src/shared.ts
@@ -1,4 +1,8 @@
 export type GriffelStylesCSSValue = string | 0;
 
 export type ValueOrArray<T> = T | Array<T>;
+/**
+ * instead of using `CSS.PropertiesFallback` from csstype that remaps properties to `readonly` union we need to use our cutom Fallback mapped type in order to avoid type issues
+ * @see https://github.com/frenic/csstype/blob/master/src/typescript.ts#L73
+ */
 export type Fallback<T> = { [P in keyof T]: T[P] | NonNullable<T[P]>[] };

